pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,385 @@
+package node
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const ManageNodeCommandName = "manage-node"
+
+type NodeConfig struct {
+	CmdFactory *clientcmd.Factory
+	Kclient    *client.Client
+	Writer     io.Writer
+
+	// Supported operations
+	Schedulable bool
+	Evacuate    bool
+	ListPods    bool
+
+	// Optional arguments
+	Selector    string
+	PodSelector string
+	DryRun      bool
+}
+
+const manageNodeLongDesc = `Manage node operations: schedulable, evacuate and list-pods.
+
+Examples:
+
+	# Block the node accepting any new pods
+	$ %[1]s manage-node <mynode> --schedulable=false
+
+	# Move selected pods to a different node
+	$ %[1]s manage-node <mynode> --evacuate --pod-selector="service=myapp"
+
+	# List all pods on nodes
+	$ %[1]s manage-node <mynode1> <mynode2> --list-pods
+
+	# Mark selected nodes as schedulable
+	$ %[1]s manage-node --selector="env=dev" --schedulable=true
+`
+
+func NewCommandManageNode(f *clientcmd.Factory, fullName, commandName string, out io.Writer) *cobra.Command {
+	cfg := NewNodeConfig()
+
+	cmd := &cobra.Command{
+		Use:   commandName,
+		Short: "Manage node operations: schedulable, evacuate, list-pods",
+		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			err := cfg.RunManageNode(f, out, c, args)
+			kcmdutil.CheckErr(err)
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.BoolVar(&cfg.Schedulable, "schedulable", true, "Control pod schedulability on the node.")
+	flags.BoolVar(&cfg.Evacuate, "evacuate", true, "Migrate all/selected pods on the node.")
```

Comment:
Make evacuate default to false, then when someone provides --evacuate it sets it to true.

----- Original Message -----
> > +func NewCommandManageNode(f *clientcmd.Factory, fullName, commandName
> > string, out io.Writer) *cobra.Command {
> > +	cfg := NewNodeConfig()
> > +
> > +	cmd := &cobra.Command{
> > +		Use:   commandName,
> > +		Short: "Manage node operations: schedulable, evacuate, list-pods",
> > +		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
> > +		Run: func(c *cobra.Command, args []string) {
> > +			err := cfg.RunManageNode(f, out, c, args)
> > +			kcmdutil.CheckErr(err)
> > +		},
> > +	}
> > +	flags := cmd.Flags()
> > +
> > +	flags.BoolVar(&cfg.Schedulable, "schedulable", true, "Control pod
> > schedulability on the node.")
> > +	flags.BoolVar(&cfg.Evacuate, "evacuate", true, "Migrate all/selected pods
> > on the node.")
> 
> When user specifies --evacuate, i want to treat that as --evacuate=true
> How are we handling this in other cli cmds? do we enforce user to provide
> either true/false always?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2151/files#r30073329